### PR TITLE
Improve default loading.

### PR DESCRIPTION
**Pull request by brendandburns**
This now is similar to `kubectl` and other libraries...

@drubin

### CODE REVIEW

File: kubernetes/src/kubernetes/config/incluster_config.rb [hunk 1]
```diff
@@ -36,6 +36,10 @@ def validate
       raise ConfigError.new("Service token file does not exists") unless File.file?(self.ca_cert)
     end
 
+    def self.is_in_cluster()
```
Comment (drubin) [inline kubernetes/src/kubernetes/config/incluster_config.rb]: This should be `def in_cluster?` its pretty un-ruby like to use `is`  methods.

File: kubernetes/lib/kubernetes/configuration.rb [hunk 2]
```diff
@@ -153,6 +153,35 @@ def self.default
...
+      end
+      # default home location
+      kc = "#{ENV['HOME']}/.kube/config"
+      if File.exist?(kc)
+        k_config = KubeConfig.new(kc)
+        k_config.configure(result)
+        return result
+      end
+      # In cluster config
+      if InClusterConfig::is_in_cluster()
```
Comment (drubin) [inline kubernetes/lib/kubernetes/configuration.rb]: same here `InClusterConfig.in_cluster?`

Also you can omit the `()` if there are no params.

Maybe we should add a linter to this too. I will add a ticket.

File: kubernetes/lib/kubernetes/configuration.rb [hunk 3]
```diff
@@ -153,6 +153,35 @@ def self.default
       @@default ||= Configuration.new
     end
 
+    def self.default_config()
```
Comment (drubin) [inline kubernetes/lib/kubernetes/configuration.rb]: Wonder if we should copy the tests from https://github.com/kubernetes-client/javascript/blob/master/src/config_test.ts#L866

File: kubernetes/src/kubernetes/config/incluster_config.rb (hunk 1 above)
Comment (brendandburns) [reply kubernetes/src/kubernetes/config/incluster_config.rb]: done.

File: kubernetes/lib/kubernetes/configuration.rb (hunk 2 above)
Comment (brendandburns) [reply kubernetes/lib/kubernetes/configuration.rb]: done.

File: kubernetes/lib/kubernetes/configuration.rb (hunk 3 above)
Comment (brendandburns) [reply kubernetes/lib/kubernetes/configuration.rb]: added.

**Comment by brendandburns** [comment]
Comments addressed, please take another look.

**Pull request merged**